scripts/poms/sbndpoms_caf_concatenation.sh

```diff
@@ -0,0 +1,290 @@
+  # SAM needs the file to have a unique name
+  ifdh addOutputFile $CONCATFILE
+  ifdh renameOutput unique
+
+  # Bit annoying but we now need to find the fcl file again as ifdh doesn't tell us what the unique name is
+  if [[ `find $WORKDIR -name "${CONCATNAME%.*}*.root" | wc -l` -ne 1 ]]
+  then
+    echo "Found incorrect number of matching files for pattern: ${CONCATNAME%.*}*.root"
+    find $WORKDIR -name "${CONCATNAME%.*}*.root"
+    echo "Exiting"
+    exit 3
+  else
+    UNIQUEOUTCONCATNAME=`find $WORKDIR -name "${CONCATNAME%.*}*.root"`
+    UNIQUEOUTCONCATNAME=`basename $UNIQUEOUTCONCATNAME`
+    echo "$CONCATNAME renamed to $UNIQUEOUTCONCATNAME"
+  fi
+
+  # OK so it looks like there is exactly one pattern match, so assume that is the correct one
+  #Copy the file to the output directory (most likely dcache)
+  ifdh copyBackOutput $OUTDIR
+  #Clear up
+  ifdh cleanup
+
+  sbndpoms_metadata_extractor.sh "$WORKDIR/$UNIQUEOUTCONCATNAME"
+
+  samweb -e sbnd declare-file $JSONFILE
+  samweb -e sbnd add-file-location ${UNIQUEOUTCONCATNAME} $OUTDIR
+  echo "$UNIQUEOUTCONCATNAME declared and located by SAM"
+
+  if [ ! -z "$FLATTEN" ]
+  then
+    echo "Flattening $UNIQUEOUTCONCATNAME"
+
+    FLATNAME="flat_caf_${SLICENUM}.root"
+    FLATFILE="$WORKDIR/$FLATNAME"
+    FLATJSONFILE="${FLATFILE}.json"
+    flatten_caf $OUTDIR/$UNIQUEOUTCONCATNAME $FLATFILE
+
+    extractCAFMetadata "$FLATFILE" > "$FLATJSONFILE"
+
+    # Steal the ifdh and sam commands from sbndpoms_genfclwithrunnumber_maker.sh (Thanks Dom)
+    # SAM needs the file to have a unique name
+    ifdh addOutputFile $FLATFILE
+    ifdh renameOutput unique
+
+    # Bit annoying but we now need to find the fcl file again as ifdh doesn't tell us what the unique name is
+    if [[ `find $WORKDIR -name "${FLATNAME%.*}*.root" | wc -l` -ne 1 ]]
+    then
+      echo "Found incorrect number of matching files for pattern: ${FLATNAME%.*}*.root"
+      find $WORKDIR -name "${FLATNAME%.*}*.root"
+      echo "Exiting"
+      exit 3
+    else
+      UNIQUEOUTFLATNAME=`find $WORKDIR -name "${FLATNAME%.*}*.root"`
+      UNIQUEOUTFLATNAME=`basename $UNIQUEOUTFLATNAME`
+      echo "$FLATNAME renamed to $UNIQUEOUTFLATNAME"
+    fi
+
+    # OK so it looks like there it exactly one pattern match, so assume that is the correct one
+    #Copy the file to the output directory (most likely dcache)
+    ifdh copyBackOutput $OUTDIR
+    #Clear up
+    ifdh cleanup
+
+    sbndpoms_metadata_extractor.sh "$WORKDIR/$UNIQUEOUTFLATNAME"
+
+    samweb -e sbnd declare-file "$FLATJSONFILE"
+    samweb -e sbnd add-file-location ${UNIQUEOUTFLATNAME} $OUTDIR
+    echo "$UNIQUEOUTFLATNAME declared and located by SAM"
+  fi
+}
+
+prepare
+
+# Loop over all of the files in the dataset
+DEFSIZE=$(samweb -e sbnd list-definition-files --summary $DEFNAME | grep "Total size" | tr -dc '0-9')
+FILECOUNT=$(samweb -e sbnd list-definition-files --summary $DEFNAME | grep "File count" | tr -dc '0-9')
```

Comment:
`samweb count-definition-files`. Not sure I have an alternative for the size though.